stablehlo/dialect/StablehloOps.td

```diff
@@ -1423,19 +1423,17 @@ def StableHLO_CompareOp: StableHLO_Op<"compare", [Pure, SameOperandsElementType,
     SameOperandsAndResultShape, Elementwise, InferTensorTypeWithReify]> {
   let summary = "Comparison operator";
   let description = [{
-    Compares `lhs` and `rhs` elementwise according to `comparison_direction`
-    and `compare_type`. If unspecified, `compare_type` is FLOAT for float element
-    types, SIGNED for signed element types and UNSIGNED for unsigned element
-    types.
+    Performs element-wise comparison of `lhs` and `rhs` tensors based on
+    `comparison_direction` and produces a `result` tensor.
 
-    See
-    https://www.tensorflow.org/xla/operation_semantics#element-wise_comparison_operations.
+    See:
+    https://github.com/openxla/stablehlo/blob/main/docs/spec_draft.md#stablehlocompare
 
     Example:
-
     ```mlir
-    %0 = stablehlo.compare LT, %arg0, %arg1 : (tensor<2xi32>, tensor<2xi32>) -> tensor<2xi1>
-    %1 = stablehlo.compare LT, %arg0, %arg1, TOTALORDER : (tensor<2xi32>, tensor<2xi32>) -> tensor<2xi1>
+    %result = "stablehlo.compare"(%lhs, %rhs) {
+      comparison_direction = #stablehlo<comparison_direction LT>
+    } : (tensor<2xf32>, tensor<2xf32>) -> tensor<2xi1>
     ```
   }];
   let arguments = (ins
```

Comment:
in line 1445 below `let results = (outs HLO_PredTensor);`
result has not alias $result

Reply:
Should I go ahead and add that?

Reply:
I'm in favor of single-output ops using `$result` but not sure if we should proactively add it without a good use case. I've hit some integration snags adding `$result` with the [`getResult`](https://mlir.llvm.org/doxygen/classmlir_1_1Operation.html#a56f58b55c803b3313da7b4a04a3d542d) function, now that we have prefixed accessors. The old function gets shadowed on the op.

Come to think of it - I'm not sure why this method gets shadowed since it has a different signature, I'll add that to my list of things to investigate eventually.. After adding `$result`, it should be that `getResult() == getResult(0)`.

Reply:
"I'm not sure why this method gets shadowed since it has a different signature". There's already `getResult()` in the [`OneResult`](https://github.com/llvm/llvm-project/blob/f62d8a1a5044df7b8d72033d056375b4ab256012/mlir/include/mlir/IR/OpDefinition.h#L610) trait upstream.

Reply:
got it, so no functional benefit to adding `$result` here, unless we needed to use it in pretty printing which isn't the case with `compare`. Probably best to not add it then